crates/types/src/blockchain/block.rs

```diff
@@ -88,12 +91,16 @@ impl Block<Transaction> {
         transactions: Vec<Transaction>,
         outbox_message_ids: &[MessageId],
         event_inbox_root: Bytes32,
-    ) -> Self {
```

Comment:
I don't think `new` should return `Result` here. Can we just pass it a valid `BlockHeader` instead of a `PartialBlockHeader`?

Reply:
Passing the block header still requires verification that transactions match the header=)

Reply:
Ah. That's right. Okay, I guess it's unavoidable. 

I think `new` gives the wrong impression still. Maybe `construct`? I just think of `new` as being a dumb constructor generally, but it's not that important. I'm fine with this.